examples/threading/threading.c

```diff
@@ -0,0 +1,74 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+	int rc;
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    usleep(thread_func_args->wait_to_obtain_ms * 1000);
+    
+    rc = pthread_mutex_lock(thread_func_args->mutex);
+    
+    if(rc){
+    ERROR_LOG("Mutex Acquire Failed\n");
+    return NULL;
+    }
+    
+    usleep(thread_func_args->wait_to_release_ms * 1000);
```

Comment:
@pranjalgupta1901 is `usleep` guaranteed to wait the specified amount of time here?